tests/strategies/order_strategy/test_order_strategy.py

```diff
@@ -38,15 +40,71 @@ def test_get_default_order_for_reinforcements(self):
         # 3. Verify expectations
         assert _result == _expected_result
 
-    def test_get_strategy_order_for_reinforcements(
+    def test_get_strategy_order_given_example_returns_default_order(
         self,
+        example_strategy_input: StrategyInput,
     ):
         # 1. Define test data.
         _expected_result = self._default_reinforcements
         _strategy = OrderStrategy()
 
         # 2. Run test.
-        _reinforcements = _strategy.get_strategy_order_for_reinforcements()
+        _reinforcements = _strategy.get_strategy_order_for_reinforcements(
+            example_strategy_input.strategy_reinforcements
+        )
+
+        # 3. Verify expectations
+        assert _reinforcements == _expected_result
+        assert _reinforcements[-1] == CofferdamReinforcementProfile
+
+    @pytest.mark.parametrize(
+        "idx", range(len(_default_reinforcements) - 1), ids=_default_reinforcements[:-1]
+    )
+    def test_get_strategy_order_increased_cost_filters_reinforcement(
```

Comment:
It is not clear to me what are you trying to validate in both tests, please add comments explaining what your expectation is and why should step 2 (run the test data) return the expected value.

Reply:
Added comments